src/test/java/uk/gov/pay/connector/gateway/worldpay/wallets/WorldpayWalletAuthorisationHandlerTest.java

```diff
@@ -110,16 +111,41 @@ public void shouldSendGooglePay3DSRequestWhenGooglePayDetailsWithoutIpAddressAre
         }
     }
 
+    @Test
+    public void shouldSendGooglePay3DSRequestWhenGooglePayDetailsWithIpAddressArePresent() throws Exception {
+        try {
+            worldpayApplePayAuthorisationHandler.authorise(getGooglePay3DSAuthorisationRequest(true, true));
+        } catch (GatewayErrorException e) {
+            verify(mockGatewayClient).postRequestFor(eq(WORLDPAY_URL), eq(gatewayAccountEntity), gatewayOrderArgumentCaptor.capture(), headers.capture());
+            assertXMLEqual(TestTemplateResourceLoader.load(WORLDPAY_VALID_AUTHORISE_WORLDPAY_GOOGLE_PAY_3DS_REQUEST_WITH_IP_ADDRESS), gatewayOrderArgumentCaptor.getValue().getPayload());
+            assertThat(headers.getValue().size(), is(1));
+            assertThat(headers.getValue(), is(AuthUtil.getGatewayAccountCredentialsAsAuthHeader(gatewayAccountEntity)));
+        }
+    }
+
+    @Test
+    public void shouldSendGooglePay3DSRequestWhenGooglePayDetailsWithout3dsEnabledArePresent() throws Exception {
```

Comment:
Can we use `3ds` instead of `3DS` for consistency?

```suggestion
    public void shouldSendGooglePay3dsRequestWhenGooglePayDetailsWithout3dsEnabledArePresent() throws Exception {
```

Reply:
Done.